package NYT.NJobTrackerClient.NProto;

import "yt/core/misc/error.proto";
import "yt/core/misc/guid.proto";
import "yt/ytlib/node_tracker_client/node.proto";

////////////////////////////////////////////////////////////////////////////////

// Describes a job to be run at a node.
message TJobSpec
{
    required int32 type = 1;

    extensions 100 to max;
}

message TStatistics
{
    message TSummary
    {
        required int64 min = 1;
        required int64 max = 2;
        required int64 sum = 3;
        required int64 count = 4;
        required string path = 5;
    }

    repeated TSummary summaries = 1;
}

// Describes the outcome of the job, in particular if it has finished successfully.
message TJobResult
{
    required NYT.NProto.TError error = 1;
    // Deprecated = 2.
    optional int64 prepare_time = 3;
    optional int64 exec_time = 4;
    required TStatistics statistics = 5;

    extensions 100 to max;
}

// Describes the current job status.
message TJobStatus
{
    required NYT.NProto.TGuid job_id = 1;
    required int32 job_type = 2;
    required int32 state = 3;
    required int32 phase = 4;
    required double progress = 5;
    optional TJobResult result = 6;
    optional NNodeTrackerClient.NProto.TNodeResources resource_usage = 7;
}

////////////////////////////////////////////////////////////////////////////////

